strategy/exception/default_exception_formatting_strategy.go

```diff
@@ -133,22 +135,25 @@ func (dEFS *DefaultFormattingStrategy) ExceptionFromError(err error) Exception {
 		Remote:  isRemote,
 	}
 
-	if err, ok := err.(*XRayError); ok {
-		e.Type = err.Type
+	xRayErr := &XRayError{}
+	if goerrors.As(err, &xRayErr) {
```

Comment:
not sure what is the reasoning behind this, but can we do it like below?

```
if goerrors.As(err, &XRayError{}) {
		e.Type = xRayErr.Type
}
```
or 
```
var xRayErr XRayError
if goerrors.As(err, &xRayErr}) {
		e.Type = xRayErr.Type
}
```

Reply:
The `err` was previously casted to a pointer type (`*XRayError`), so to preserve that behavior an address of a pointer is passed down to `errors.As`.
The first suggestion seems to be incorrect (`xRayErr` variable undefined), the second leaves us with a value, not a pointer.